File: Python/50. Pow(x, n).py
class Solution:
    def myPow(self, x: float, n: int) -> float:
#         return pow(x, n)
#         return x ** n

        if n < 0:
            x = 1 / x
            n = abs(n)
        result = 1
        while n > 0:
            if n & 1:
                result = result * x
            x = x * x
            n >>= 1
        return result

File: C/50. Pow(x, n).c
double pow_(double x, long long n) {
    if (n == 0) return 1;
    if (n < 0) return 1.0 / pow(x, n * -1);

    if (n % 2) return x * pow(x * x, n / 2);
    else return pow(x * x, n / 2);
}

double myPow(double x, int n){
    return pow_(x, (long long)n);
}
